zappa/core.py

```diff
@@ -918,15 +918,17 @@ def update_lambda_configuration(    self,
         if not aws_environment_variables:
             aws_environment_variables = {}
         else:
-            # Get any remote aws lambda env vars so they don't get trashed
-            # Related: https://github.com/Miserlou/Zappa/issues/765
-            lambda_aws_environment_variables = self.lambda_client\
-                .get_function_configuration(FunctionName=function_name)["Environment"].get("Variables", {})
-
-            # Append keys that are remote but not in settings file
-            for key, value in lambda_aws_environment_variables.items():
-                if key not in aws_environment_variables:
-                    aws_environment_variables[key] = value
+            # There may not be an Environment key defined.
+            # https://github.com/Miserlou/Zappa/issues/987
+            lambda_aws_environment_variables = self.lambda_client.get_function_configuration(FunctionName=function_name)
+            if "Environment" in lambda_aws_environment_variables:
+                # Get any remote aws lambda env vars so they don't get trashed
+                # Related: https://github.com/Miserlou/Zappa/issues/765
+                lambda_aws_environment_variables["Environment"].get("Variables", {})
```

Comment:
and probably rename the lambda_aws_environment_variables on line 923 since it is now all of the function config, not just the environment